clusterloader2/pkg/measurement/common/slos/slo_measurement.go

```diff
@@ -0,0 +1,140 @@
+
+	var sloMeasurement sloMeasurement
+	var err error
+	
+	if sloMeasurement.networkProgrammingLatency, err = measurement.CreateMeasurement("NetworkProgrammingLatency"); err != nil {
+		klog.Errorf("cannot create instance of %s: %v", "NetworkProgrammingLatency", err)
+	}
+	if sloMeasurement.dnsLookupLatency, err = measurement.CreateMeasurement("DnsLookupLatency"); err != nil {
+		klog.Errorf("cannot create instance of %s: %v", "DnsLookupLatency", err)
+	}
+
+	createSloMeasurement := func() measurement.Measurement {
+		return createSloMeasurement(sloMeasurement.networkProgrammingLatency, sloMeasurement.dnsLookupLatency)
+	}
+	if err := measurement.Register(sloMeasurementName, createSloMeasurement); err != nil {
+		klog.Fatalf("Cannot register %s: %v", sloMeasurementName, err)
+	}
+}
+
+func createSloMeasurement(networkProgrammingLatency measurement.Measurement, dnsLookupLatency measurement.Measurement) measurement.Measurement {
+	return &sloMeasurement{
+		networkProgrammingLatency: networkProgrammingLatency,
+		dnsLookupLatency:          dnsLookupLatency,
+	}
+}
+
+type sloMeasurement struct {
+	dnsLookupLatency                 measurement.Measurement
+	networkProgrammingLatency        measurement.Measurement
+	summaries                        []measurement.Summary
+	dnsLookupLatencyMetrics          *sloMeasurementOutput
+	networkProgrammingLatencyMetrics *sloMeasurementOutput
+}
+
+type sloMeasurementOutput struct {
+	DnsLookupLatencyMetrics          []measurement.Summary `json:"dnsLookupLatencyMetrics"`
+	NetworkProgrammingLatencyMetrics []measurement.Summary `json:"networkProgrammingLatencyMetrics"`
+}
+
+func (s *sloMeasurement) start(config *measurement.Config) error {
+	_, err := s.dnsLookupLatency.Execute(config)
+	if err != nil {
+		klog.Errorf("Error starting %s: %v", "DnsLookupLatency", err)
+		return err
+	}
+	_, err = s.networkProgrammingLatency.Execute(config)
+	if err != nil {
+		klog.Errorf("Error starting %s: %v", "NetworkProgrammingLatency", err)
+		return err
+	}
+	return nil
+}
+
+func (s *sloMeasurement) gather(config *measurement.Config) error{
+	output := sloMeasurementOutput{}
+	errList := errors.NewErrorList()
+
+	dnsLookupSummary, err := s.dnsLookupLatency.Execute(config)
+	if err != nil {
+		errList.Append(errList, err)
+	} else {
+		output.DnsLookupLatencyMetrics = dnsLookupSummary
+	}
+
+	networkProgrammingLatencySummary, err := s.networkProgrammingLatency.Execute(config)
+	if err != nil {
+		errList.Append(errList, err)
+	} else {
+		output.NetworkProgrammingLatencyMetrics = networkProgrammingLatencySummary
+	}
+
+	content, err := util.PrettyPrintJSON(output)
+	if err != nil {
+		errList.Append(errList, err)
+	} else {
+		summary := measurement.CreateSummary(sloMeasurementName, "json", content)
+		s.summaries = append(s.summaries, summary)
+	}
+	return errList
+}
+
+func (s sloMeasurement) Execute(config *measurement.Config) ([]measurement.Summary, error) {
+	action, err := util.GetString(config.Params, "action")
```

Comment:
You don't need all this logic - there is no need to parse it.

Basically all you need to do is:
```
for _, m := range s.measurements {
  m.Execute(config)
}
```

You also don't need start, gather methods here - this should be simply delegated to underlying methods (since that allows us to implement more methods without changing this measurement).

Reply:
I did that. The only doubt is, do I collect the summaries/errors provided by each of the underlying measurements and return them as part of SLOMeasurement? Because, the execute method of SLOMeasurement also expects summaries and error returned.